RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+            break
+        }
+
+        // Clear the promotional offer data to dismiss the sheet
+        self.promotionalOfferData = nil
+
+        if userAction.shouldTerminateCurrentPathFlow {
+            self.loadingPath = nil
+        } else {
+            if let loadingPath = loadingPath {
+                await self.onPathSelected(path: loadingPath)
+                self.loadingPath = nil
+            }
+        }
+    }
+}
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+private extension BaseManageSubscriptionViewModel {
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    // swiftlint:disable:next cyclomatic_complexity
+    private func onPathSelected(path: CustomerCenterConfigData.HelpPath) async {
+        switch path.type {
+        case .missingPurchase:
+            self.showRestoreAlert = true
+
+        case .refundRequest:
+            guard let purchaseInformation = self.purchaseInformation else { return }
+            let productId = purchaseInformation.productIdentifier
+            do {
+                self.actionWrapper.handleAction(.refundRequestStarted(productId))
+
+                let status = try await self.purchasesProvider.beginRefundRequest(forProduct: productId)
+                self.refundRequestStatus = status
+                self.actionWrapper.handleAction(.refundRequestCompleted(productId, status))
+            } catch {
+                self.refundRequestStatus = .error
+                self.actionWrapper.handleAction(.refundRequestCompleted(productId, .error))
+            }
+
+        case .cancel where purchaseInformation?.store != .appStore:
+            if let url = purchaseInformation?.managePurchaseURL {
+                self.inAppBrowserURL = IdentifiableURL(url: url)
+            } else {
+                // WIP: shor error alert?
+            }
+
+        case .changePlans, .cancel:
+            self.actionWrapper.handleAction(.showingManageSubscriptions)
+
+        case .customUrl:
+            guard let url = path.url,
+                  let openMethod = path.openMethod else {
+                Logger.warning("Found a custom URL path without a URL or open method. Ignoring tap.")
+                return
+            }
+            switch openMethod {
+            case .external,
+                _ where !url.isWebLink:
+                URLUtilities.openURLIfNotAppExtension(url)
+            case .inApp:
+                self.inAppBrowserURL = .init(url: url)
+            @unknown default:
+                Logger.warning(Strings.could_not_determine_type_of_custom_url)
+                URLUtilities.openURLIfNotAppExtension(url)
+            }
+
+        default:
+            break
+        }
+    }
+#endif
+
+}
+
+private extension CustomerCenterConfigData.Screen {
+
+    var filteredPaths: [CustomerCenterConfigData.HelpPath] {
```

Comment:
nitpick: maybe something like `supportedPaths`? `filtered` doesn't really tell you what it's filtering for at a glance